src/Voron/StorageEnvironment.cs

```diff
@@ -600,6 +601,17 @@ public Transaction WriteTransaction(ByteStringContext context = null, TimeSpan?
             return writeTransaction;
         }
 
+        public LowLevelTransaction SafeNewLowLevelTransaction(
+            TransactionPersistentContext transactionPersistentContext,
+            TransactionFlags transactionFlags,
+            TimeSpan? timeout = null)
+        {
+            using (MemoryInformation.DisableOutOfMemoryChecks())
```

Comment:
This way we disable the check but still do the allocation - isn't it dangerous?

Reply:
Providing the API as suggested in https://github.com/ravendb/ravendb/pull/7989#discussion_r232018490 is more complex but would guarantee we won't do the allocation

Reply:
We do allocate 4KB for this and we don't do any significant work (write do write a single page or just hold the write lock).

Reply:
Where is this allocation - `LowLevelTransaction.InitTransactionHeader` ?

Reply:
I wonder if we could mark that this write tx isn't going to be committed (and throw on attempt to commit) and then don't do this allocation

Reply:
The problem is that we have two cases where we create a write tx mostly to keep the write tx lock but we also do the commit:

1) WriteAheadJournal.WaitForJournalStateToBeUpdated

https://github.com/ravendb/ravendb/blob/54cc370bfeecf2bf1bb8bca27718a2d7d945525e/src/Voron/Impl/Journal/WriteAheadJournal.cs#L709-L713

2) ScratchBufferPool.Cleanup
https://github.com/ravendb/ravendb/blob/54cc370bfeecf2bf1bb8bca27718a2d7d945525e/src/Voron/Impl/Scratch/ScratchBufferPool.cs#L460-L463

Although in first case I think we have the commit because of possible lazy tx mode usage?
https://github.com/ravendb/ravendb/blob/54cc370bfeecf2bf1bb8bca27718a2d7d945525e/src/Voron/Impl/Journal/WriteAheadJournal.cs#L764-L767

Reply:
Maybe since the allocation potentially causing problems in low memory is made in `InitTransactionHeader` and we don't always need it, we use it only if we write to journal:

https://github.com/ravendb/ravendb/blob/6c7905bdcd1f5c0b84c2be0371adc68a88523a77/src/Voron/Impl/Journal/WriteAheadJournal.cs#L1467

then maybe we could do this allocation on demand (lazily) instead of calling it in the ctor of LowLevelTransaction